libs/@guardian/libs/src/consent-management-platform/usnat/getConsentState.ts

```diff
@@ -1,18 +1,29 @@
 import { type USNATConsentState } from '../types/usnat';
 import { getGPPData } from './api';
 
+const getSupportedAPIv1 = (
+	supportedAPI: string | undefined,
+): string | undefined => {
+	if (supportedAPI?.includes('usnat')) {
+		return supportedAPI.includes('v1') ? supportedAPI : `${supportedAPI}v1`;
+	}
+	return supportedAPI;
+};
+
 export const getConsentState: () => Promise<USNATConsentState> = async () => {
 	let doNotSell = false; // Opt-Out
 	const gppData = await getGPPData();
 
-	const supportedAPIs = gppData.supportedAPIs[0]?.split(':')[1]; // E.G: '7:usnatv1', '8:uscav1'
+	const supportedAPI = gppData.supportedAPIs[0]?.split(':')[1]; // E.G: '7:usnatv1', '8:uscav1'
+	// Temporary fix 21/11/2024
+	const supportedAPIv1 = getSupportedAPIv1(supportedAPI);
 
-	if (supportedAPIs) {
-		//0 Not Applicable. SharingOptOutNotice value was not applicable or no notice was provided, 1 Opted Out, 2 Did Not Opt Out
-		doNotSell =
-			gppData.parsedSections[supportedAPIs]?.SaleOptOut !== 2 ||
-			gppData.parsedSections[supportedAPIs].Gpc;
+	if (supportedAPIv1) {
 		// https://github.com/InteractiveAdvertisingBureau/Global-Privacy-Platform/blob/main/Sections/US-National/IAB%20Privacy%E2%80%99s%20National%20Privacy%20Technical%20Specification.md
+		// 0 Not Applicable. SharingOptOutNotice value was not applicable or no notice was provided, 1 Opted Out, 2 Did Not Opt Out
+		doNotSell =
+			gppData.parsedSections[supportedAPIv1]?.SaleOptOut !== 2 ||
```

Comment:
is there any issue with doing `gppData.parsedSections['usnatv1']?.SaleOptOut` instead?

Reply:
I think we saw when we were debugging that this code ran for all the frameworks so didn't want to hardcode to `usnat`.

Reply:
There are 5 states that have their own id's/frameworks. 'usnat', 'usca','usva', 'usco', 'usut', 'usct'

Reply:
hence, we need to check the supportedAPis to check the framework. There are other states that might follow these 5 in the future so it has be dynamically checked.